homeassistant/components/youless/sensor.py

```diff
@@ -0,0 +1,213 @@
+        self._gateway.update()
+
+    @property
+    def get_sensor(self) -> Optional[YoulessSensor]:
+        """Property to get the underlying sensor object."""
+        return None
+
+    @property
+    def unit_of_measurement(self) -> Optional[str]:
+        """Return the unit of measurement for the sensor."""
+        if self.get_sensor is None:
+            return None
+
+        return self.get_sensor.unit_of_measurement
+
+    @property
+    def state(self) -> StateType:
+        """Determine the state value, only if a sensor is initialized."""
+        if self.get_sensor is None:
+            return None
+
+        return self.get_sensor.value
+
+    @property
+    def unique_id(self) -> Optional[str]:
+        """Return the uniquely generated id."""
+        return f"{DOMAIN}_{self._device}_{self._sensor_id}"
+
+    @property
+    def available(self) -> bool:
+        """Return a flag to indicate the sensor not being available."""
+        return self.get_sensor is not None
+
+    @property
+    def device_info(self) -> Optional[Dict[str, Any]]:
+        """Return the device information."""
+        return {
+            "identifiers": {(DOMAIN, self._gateway.mac_address)},
+            "name": self._device,
+            "manufacturer": "YouLess",
+            "model": self._gateway.model,
+        }
+
+
+class GasSensor(YoulessBaseSensor):
+    """The Youless gas sensor."""
+
+    def __init__(self, gateway: YoulessAPI, device: str):
+        """Instantiate a gas sensor."""
+        super().__init__(gateway, device, "Gas meter", "gas")
+
+    @property
+    def name(self) -> Optional[str]:
+        """Return the name of the meter."""
+        return f"{self._device} gas"
+
+    @property
+    def icon(self) -> Optional[str]:
+        """Return the icon to use in the frontend, if any."""
+        return "mdi:fire"
+
+    @property
+    def get_sensor(self) -> Optional[YoulessSensor]:
+        """Get the sensor for providing the value."""
+        return self._gateway.gas_meter
+
+
+class CurrentPowerSensor(YoulessBaseSensor):
+    """The current power usage sensor."""
+
+    def __init__(self, gateway: YoulessAPI, device: str):
+        """Instantiate the usage meter."""
+        super().__init__(gateway, device, "Power usage", "usage")
+        self._device = device
+
+    @property
+    def name(self) -> Optional[str]:
+        """Return the name of the meter."""
+        return f"{self._device} Usage"
+
+    @property
+    def get_sensor(self) -> Optional[YoulessSensor]:
+        """Get the sensor for providing the value."""
+        return self._gateway.current_power_usage
+
+    @property
+    def device_class(self) -> Optional[str]:
+        """Return the class of this device, from component DEVICE_CLASSES."""
+        return DEVICE_CLASS_POWER
+
+    @property
+    def icon(self) -> Optional[str]:
+        """Return the icon to use in the frontend, if any."""
+        return "mdi:power-socket"
```

Comment:
When a device class is provided, one should not provide an icon. It is the one or the other (more cases in this PR).